src/proto_conversions.cc

```diff
@@ -1,7 +1,52 @@
 #include "proto_conversions.h"
 
+using namespace valhalla;
+
 namespace valhalla {
 
+std::string incidentTypeToString(const TripLeg_Node_Incident& incident) {
+  std::string incident_type;
+  switch (incident.type()) {
+    case TripLeg_Node_Incident_Type_ACCIDENT:
+      incident_type = "accident";
```

Comment:
Maybe we can avoid a copy by returning directly from the `case` statements? (not sure if the compiler can do away with the string copy being created here, maybe it can?)

Reply:
Not sure either. I'd assume the compiler could optimize it, but I can make it a direct return.